### PR TITLE
Energy dagger is quieter, and can backstab

### PR DESCRIPTION
**Pull request by BiancaWilkson**
## What Does This PR Do

Makes all sounds made by the energy dagger significantly quieter. Also adds a backstab mechanic that knocks down, and deals extra damage on backstabs.
## Why It's Good For The Game

The energy dagger is a pretty boring weapon, that does not see much use at all. It's supposed to be a stealthier alternative as a weapon, but it's just as loud as an e-sword. I wanted to solve this issue, while also giving it more of an assassination flair by allowing for backstabs.
#15377 tried to do this a while ago, and that got stuck in development hell. I took most of the code straight from that PR so thank you Charlie.

##Images of Changes

https://user-images.githubusercontent.com/42818125/208564330-4c6f62f6-f7f4-4393-900b-2205d2968e3d.mp4

## Testing

I spawned in, and stabbed a whole lot of skrells in the back.

## Changelog
:cl:
tweak: Edagger is quieter, and can backstab
/:cl:

### CODE REVIEW

**Comment by SpringSkipper** [comment]
Instead of "M.apply_damage(12, BRUTE, BODY_ZONE_CHEST)", you can do...

force += 12
. = ..()
force -= 12

That would make it correctly apply all 30 damage to the same body zone, and would prevent the 12 bonus damage from completely ignoring armor.

**Comment by procdrone** [comment]
considering the assassination angle, maybe we could make the sound not travel through walls/windows? (so half the station doesn't immediately hear when you try to stab someone with a small tiny energy dagger) That would be pretty neat and niche use for this weapon which, IMO, would fit in just fine.

File: code/modules/paperwork/pen.dm [hunk 1]
```diff
@@ -147,8 +147,20 @@
...
+	stealthy_audio = TRUE
 	armour_penetration_flat = 20
 
+/obj/item/pen/edagger/attack(mob/living/M, mob/living/user, def_zone)
+	if(on && user.dir == M.dir && !M.incapacitated(TRUE) && user != M)
+		M.apply_damage(12, BRUTE, BODY_ZONE_CHEST)
+		M.apply_damage(40, STAMINA) //Just enough to slow
+		M.KnockDown(2 SECONDS)
+		M.visible_message("<span class='warning'>[user] stabs [M] in the back!</span>", "<span class='userdanger'>[user] stabs you in the back! The energy blade makes you collapse in pain!</span>")
+	. = ..()
```
Comment (hal9000PR) [inline code/modules/paperwork/pen.dm:159]:
```suggestion
/obj/item/pen/edagger/attack(mob/living/M, mob/living/user, def_zone)
	if(on && user.dir == M.dir && !HAS_TRAIT(src, TRAIT_FLOORED) && user != M)
		M.apply_damage(12, BRUTE, def_zone)
		M.apply_damage(40, STAMINA) //Just enough to slow
		M.KnockDown(2 SECONDS)
		M.visible_message("<span class='warning'>[user] stabs [M] in the back!</span>", "<span class='userdanger'>[user] stabs you in the back! The energy blade makes you collapse in pain!</span>")
	. = ..()
```

**Review by Sirryan2002 — APPROVED** [review]
code looks good

File: code/modules/paperwork/pen.dm [hunk 2]
```diff
@@ -147,8 +147,20 @@
 	var/on = FALSE
 	var/brightness_on = 2
 	light_color = LIGHT_COLOR_RED
+	stealthy_audio = TRUE
 	armour_penetration_flat = 20
 
+/obj/item/pen/edagger/attack(mob/living/M, mob/living/user, def_zone)
+	if(on && user.dir == M.dir && !HAS_TRAIT(src, TRAIT_FLOORED) && user != M)
```
Comment (SteelSlayer) [inline code/modules/paperwork/pen.dm:154]: Think you meant to check that the user or M is floored? Right now you're checking if the edagger is floored.

**Review by SteelSlayer — CHANGES_REQUESTED** [review]

File: code/modules/paperwork/pen.dm (hunk 2 above)
Comment (S34NW) [inline code/modules/paperwork/pen.dm:154]:
```suggestion
	if(on && user.dir == M.dir && !HAS_TRAIT(user, TRAIT_FLOORED) && user != M)
```

File: code/modules/paperwork/pen.dm [hunk 3]
```diff
@@ -147,19 +147,36 @@
...
-		M.visible_message("<span class='warning'>[user] stabs [M] in the back!</span>", "<span class='userdanger'>[user] stabs you in the back! The energy blade makes you collapse in pain!</span>")
-	. = ..()
-
-/obj/item/pen/edagger/get_clamped_volume()
-	return 5
+	if(on && HAS_TRAIT(user, TRAIT_PACIFISM))
+		to_chat(user, "<span class='warning'>You don't want to harm other living beings!</span>")
+		return
+	if(!on)
+		playsound(loc, 'sound/weapons/tap.ogg', get_clamped_volume(), 1, -1)
```
Comment (SteelSlayer) [inline code/modules/paperwork/pen.dm:158]: Why did you remove the parent call and start copy and pasting it's code here? What was wrong with the way it was written before?

**Review by SteelSlayer — CHANGES_REQUESTED** [review]

File: code/modules/paperwork/pen.dm (hunk 3 above)
Comment (SteelSlayer) [reply code/modules/paperwork/pen.dm:158]: Well, for the reactive armor, pretty sure you can just use the parent return value to decide whether or not to continue, since it gets set to `M.attacked_by`. With the way you have it now, the backstab damage is always applied no matter what, given the dagger is on. Also the parent `playsound` call already has `ignore_walls = TRUE`.

**Review by Sirryan2002 — CHANGES_REQUESTED** [review]
rc untile steel's requests are met

File: code/modules/paperwork/pen.dm [hunk 4]
```diff
@@ -147,8 +147,26 @@
 	var/on = FALSE
 	var/brightness_on = 2
 	light_color = LIGHT_COLOR_RED
+	var/backstab_sound = 'sound/items/unsheath.ogg'
 	armour_penetration_flat = 20
 
+/obj/item/pen/edagger/attack(mob/living/M, mob/living/user, def_zone)
+	force = 18
+	if(user.dir == M.dir && !HAS_TRAIT(M, TRAIT_FLOORED) && user != M)
+		force += 12
```
Comment (hal9000PR) [inline code/modules/paperwork/pen.dm:156]: I dont like setting the force to 18 here. 

I am pretty sure htis means if you bap people with the pen in pen mode it will brutally hit them for 18 damage.

what if you had a var on the proc that was like `var/applied_extra_force = FALSE` then set it to true when you increase the force. then check after the parent call if you increased the force, then decrease it.

also please make the `12` a variable. like `backstab damage`

File: code/modules/paperwork/pen.dm [hunk 5]
```diff
@@ -147,8 +147,32 @@
 	var/on = FALSE
 	var/brightness_on = 2
 	light_color = LIGHT_COLOR_RED
+	var/backstab_sound = 'sound/items/unsheath.ogg'
+	var/backstab_damage = 12
+	var/extra_force_applied = FALSE
```
Comment (farie82) [inline code/modules/paperwork/pen.dm:152]:
```suggestion
```
This can be a local variable instead. No need to reset it then afterwards

File: code/modules/paperwork/pen.dm [hunk 6]
```diff
@@ -147,8 +147,32 @@
...
 	armour_penetration_flat = 20
 
+/obj/item/pen/edagger/attack(mob/living/M, mob/living/user, def_zone)
+	if(on && user.dir == M.dir && !HAS_TRAIT(M, TRAIT_FLOORED) && user != M)
```
Comment (farie82) [inline code/modules/paperwork/pen.dm:156]:
```suggestion
	var/extra_force_applied = FALSE
	if(on && user.dir == M.dir && !HAS_TRAIT(M, TRAIT_FLOORED) && user != M)
```

File: code/modules/paperwork/pen.dm [hunk 7]
```diff
@@ -147,8 +147,32 @@
...
+		M.KnockDown(2 SECONDS)
+		M.visible_message("<span class='warning'>[user] stabs [M] in the back!</span>", "<span class='userdanger'>[user] stabs you in the back! The energy blade makes you collapse in pain!</span>")
+
+		playsound(loc, backstab_sound, 5, TRUE, ignore_walls = FALSE, falloff_distance = 0)
+	else
+		playsound(loc, hitsound, 5, TRUE, ignore_walls = FALSE, falloff_distance = 0)
+	. = ..()
+	if(extra_force_applied)
+		force -= backstab_damage
+		extra_force_applied = FALSE
```
Comment (farie82) [inline code/modules/paperwork/pen.dm:170]:
```suggestion
```

**Review by farie82 — APPROVED** [review]

**Review by Sirryan2002 — APPROVED** [review]

**Review by SteelSlayer — APPROVED** [review]
